### PR TITLE
Set swapRB to false in Caffe tests and examples.

### LINKED ISSUE #9625: Caffe models are trained in BGR image format, not RGB
Some of samples with Caffe models wrongly swap red and blue channels (by keeping swapRB argument of `blobFromImage` with default `true` value). However models are trained in BGR input format, see https://github.com/BVLC/caffe/wiki/Image-Format:-BGR-not-RGB.

source: http://answers.opencv.org/question/174139/is-the-swaprb-value-in-the-example-googlenet-dnn-code-wrong/

GoogLeNet image classification using DNN:

Image | RGB input (class id / confidence) | BGR input 
----------|--------------|------------
space shuttle <br>  | 812 0.999828 | 812 0.999935
red wine <br>  | 966 0.869604 | 966 **0.995303**
jeans <br>  | 608 0.83024 | 608 **0.998637**

We need to find the all Caffe networks are used in our samples / tutorials and check the correctness.

### PR DESCRIPTION
**Pull request by jrobble**
Resolves #9625 

**Merge with contrib**: https://github.com/opencv/opencv_contrib/pull/1372
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/378

Set swapRB to false in Caffe tests and examples.

### CODE REVIEW

File: samples/dnn/mobilenet_ssd_python.py [hunk 1]
```diff
@@ -23,25 +23,24 @@
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument("--video", help="path to video file. If empty, camera's stream will be used")
-    parser.add_argument("--prototxt", default="MobileNetSSD_deploy.prototxt",
+    parser.add_argument("--prototxt", default="MobileNetSSD_300x300.prototxt",
```
Comment (dkurt) [inline samples/dnn/mobilenet_ssd_python.py]: Please, keep actual changes of https://github.com/opencv/opencv/pull/9333.

**Comment by dkurt** [comment]
@jrobble, Thank you! Please, update also these files:
- [ ] https://github.com/opencv/opencv/blob/master/samples/android/mobilenet-objdetect/src/org/opencv/samples/opencv_mobilenet/MainActivity.java#L87
- [ ] https://github.com/opencv/opencv/blob/master/samples/dnn/resnet_ssd_face_python.py#L30

**Comment by jrobble** [comment]
@dkurt, you're welcome. I updated the files you requested. Note that in [tracker.hpp](https://github.com/opencv/opencv_contrib/blob/master/modules/tracking/include/opencv2/tracking/tracker.hpp) it states:

> GOTURN architecture goturn.prototxt and trained model goturn.caffemodel are accessible on opencv_extra GitHub repository.

Also, in [goturnTracker.cpp](https://github.com/opencv/opencv_contrib/blob/master/modules/tracking/samples/goturnTracker.cpp) it states:

> //2 - Download pretrained caffemodel from <https://github.com/opencv/opencv_extra>

I could not find goturn.prototxt or goturn.caffemodel in opencv_extra. I had to download them from a [previous commit](https://github.com/opencv/opencv_extra/tree/c4219d5eb3105ed8e634278fad312a1a8d2c182d/testdata/tracking).

**Comment by dkurt** [comment]
@jrobble, May I ask you to close these PRs and create the new ones but with different from `master` source branches names? BuildBot can't test it automatically.

**Comment by jrobble** [comment]
@dkurt, done!

**Pull request closed without merging**